pkg/diagnostics/analyzers.go

```diff
@@ -362,3 +366,33 @@ func (a *analyzerFactory) controlPlaneIPAnalyzer() []*Analyze {
 		},
 	}
 }
+
+// vmsAccessAnalyzer will analyze if vms have access to the API server of vSphere cluster
+func (a *analyzerFactory) vmsAccessAnalyzer() *Analyze {
+	runBashPod := "check-cloud-controller"
+	runBashPodLog := fmt.Sprintf("%s.log", runBashPod)
+	vSphereCloudControllerPodLogPath := path.Join(runBashPod, runBashPodLog)
+	return &Analyze{
+		TextAnalyze: &textAnalyze{
+			analyzeMeta: analyzeMeta{
+				CheckName: fmt.Sprintf("%s: VMs has no access to vSphere APi. Logs: %s", logAnalysisAnalyzerPrefix, vSphereCloudControllerPodLogPath),
+			},
+			FileName:     vSphereCloudControllerPodLogPath,
+			RegexPattern: `Failed to create new client. err: Post (.*) dial tcp (.*) connect: connection timed out\n(.*)Failed to create govmomi client. err: Post (.*) dial tcp (.*) connect: connection timed out`,
+			Outcomes: []*outcome{
+				{
+					Fail: &singleOutcome{
+						When:    "true",
+						Message: fmt.Sprintf("Failed to create client, Virtural Machines have no access to vSphere API. See the cloud controller log in master node: %s", vSphereCloudControllerPodLogPath),
```

Comment:
> I wonder if there's a relevent link to the documentation we can provide. I don't know if we have docs which lay out the required connection to Vsphere but that would be ideal.

Do you mind to tell more about which kind of doc would make this better? I probably didn't get the point when you said `required connection to vsphere` . 
My understanding is that it's reasonable if we assume that it should connect to vsphere because of that this analyzer supposes to do its job when it's confirmed that a vSphere cluster will be created.

Reply:
I'm also trying to add a warning message to this analyzer. Because the analyzer should only work on workload cluster instead of bootstrap cluster. Currently, it's not that smart to be able to figure out the condition it is woking on. Hopefully, a warning message could guide user to make a better use of it.